### PR TITLE
genesis: pass in more parameters vs using constants

### PR DESCRIPTION
**Pull request by bmwill**
- genesis: pass in stake subsidy parameters vs using constants
- genesis: derive TransactionDigest used for genesis from inputs instead of using all 0's
- config: track SuiAddress and not AccountKey in ValidatorInfo

Fixes: https://github.com/MystenLabs/sui/issues/9568

### CODE REVIEW

File: crates/sui-config/src/genesis.rs [hunk 1]
```diff
@@ -816,10 +868,26 @@ impl Builder {
...
+    genesis_validators: &[GenesisValidatorMetadata],
+    token_distribution_schedule: &TokenDistributionSchedule,
+) -> TxContext {
+    let mut hasher = DefaultHash::default();
+    hasher.update(b"sui-genesis");
+    hasher.update(&bcs::to_bytes(genesis_chain_parameters).unwrap());
+    hasher.update(&bcs::to_bytes(genesis_validators).unwrap());
+    hasher.update(&bcs::to_bytes(token_distribution_schedule).unwrap());
+    let hash = hasher.finalize();
+    let genesis_transaction_digest = TransactionDigest::new(hash.into());
```
Comment (bmwill) [inline crates/sui-config/src/genesis.rs:886]: @kchalkias Can you provide guidance on how best to do this hashing?

File: crates/sui-config/src/genesis.rs [hunk 2]
```diff
@@ -816,10 +868,26 @@ impl Builder {
     }
 }
 
-fn get_genesis_context(epoch_data: &EpochData) -> TxContext {
+// Create a Genesis Txn Context to be used when generating genesis objects by hashing all of the
+// inputs into genesis ans using that as our "Txn Digest". This is done to ensure that coin objects
```
Comment (kchalkias) [inline crates/sui-config/src/genesis.rs:872]: an**s** -> and

File: crates/sui-config/src/genesis.rs (hunk 1 above)
Comment (kchalkias) [reply crates/sui-config/src/genesis.rs:886]: Ah maybe it's not going to solve anything, just help on readability.

Comment (bmwill) [reply crates/sui-config/src/genesis.rs:886]: @kchalkias The only other input not included in this hash is the sui/move framework module bytecode. What are you feelings or thoughts for or against including that in the hash as well?

File: crates/sui-framework/sources/governance/sui_system.move [hunk 3]
```diff
@@ -45,24 +45,28 @@ module sui::sui_system {
...
     ) {
         let system_state = sui_system_state_inner::create(
             validators,
             stake_subsidy_fund,
             storage_fund,
-            governance_start_epoch,
-            initial_stake_subsidy_amount,
             protocol_version,
             system_state_version,
+            governance_start_epoch,
```
[the remark this answers is not in the document]
Comment (bmwill) [reply crates/sui-framework/sources/governance/sui_system.move:64]: Yep i have triple checked that the ordering is correct.

**Pull request merged**